Add support for latest GraalVM version

`native-image --version` looks like this for JDK 22 based build:

```
native-image 22 2024-03-19
GraalVM Runtime Environment GraalVM CE 22-dev+15.1 (build 22+15-jvmci-b01)
Substrate VM GraalVM CE 22-dev+15.1 (build 22+15, serial gc)
```

This fails many tests, one basic test would be `AppReproducersTest#versionsParsingMandrel`, which fails with this as follows:

```
2023-10-02 18:58:06.845 INFO  [o.g.t.i.u.v.UsedVersion$MVersion] (runNativeImageVersion) Running command [native-image, --version] to determine Mandrel version used.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.362 s <<< FAILURE! - in org.graalvm.tests.integration.AppReproducersTest
[ERROR] versionsParsingMandrel{TestInfo}  Time elapsed: 0.014 s  <<< ERROR!
java.lang.ExceptionInInitializerError
Caused by: java.lang.NullPointerException: Cannot invoke "String.toLowerCase()" because "version" is null

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   AppReproducersTest.versionsParsingMandrel » ExceptionInInitializer
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Native image integration TS 1.0.0-SNAPSHOT:
[INFO] 
[INFO] Native image integration TS ........................ SUCCESS [  0.094 s]
[INFO] testsuite .......................................... FAILURE [  3.620 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.785 s
[INFO] Finished at: 2023-10-02T18:58:07+02:00
[INFO] ------------------------------------------------------------------------
```

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/versions/UsedVersion.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/versions/UsedVersion.java
@@ -26,6 +26,7 @@ import org.jboss.logging.Logger;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
@@ -69,6 +70,12 @@ import static org.graalvm.tests.integration.utils.Commands.getRunCommand;
  * GraalVM Runtime Environment GraalVM CE (build 20+34-jvmci-23.0-b10)
  * Substrate VM GraalVM CE (build 20+34, serial gc)
  *
+ * or
+ *
+ * native-image 22 2024-03-19
+ * GraalVM Runtime Environment GraalVM CE 22-dev+15.1 (build 22+15-jvmci-b01)
+ * Substrate VM GraalVM CE 22-dev+15.1 (build 22+15, serial gc)
+ *
  * @author Michal Karm Babacek <karm@redhat.com>
  */
 public class UsedVersion {
@@ -96,6 +103,11 @@ public class UsedVersion {
 
     // Implements version parsing after https://github.com/oracle/graal/pull/6302
     static final class VersionParseHelper {
+
+        private static final Map<Integer, String> GRAAL_MAPPING = Map.of(22, "24.0",
+                                                                         23, "24.1",
+                                                                         24, "25.0",
+                                                                         25, "25.1");
 
         private static final String JVMCI_BUILD_PREFIX = "jvmci-";
         private static final String MANDREL_VERS_PREFIX = "Mandrel-";
@@ -145,7 +157,7 @@ public class UsedVersion {
                 String vendorVersion = secondMatcher.group(VENDOR_VERSION_GROUP);
 
                 String buildInfo = secondMatcher.group(BUILD_INFO_GROUP);
-                String graalVersion = graalVersion(buildInfo);
+                String graalVersion = graalVersion(buildInfo, v.feature());
                 String mandrelVersion = mandrelVersion(vendorVersion);
                 String versNum = (isMandrel(vendorVersion) ? mandrelVersion : graalVersion);
                 Version vers = versionParse(versNum);
@@ -191,7 +203,11 @@ public class UsedVersion {
             return null;
         }
 
-        private static String graalVersion(String buildInfo) {
+        private static String graalVersion(String buildInfo, int jdkFeatureVers) {
+            if (jdkFeatureVers >= 22) {
+                // short-circuit new version scheme with a mapping
+                return GRAAL_MAPPING.get(Integer.valueOf(jdkFeatureVers));
+            }
             if (buildInfo == null) {
                 return null;
             }

